app/src/main/java/org/wikipedia/views/SurveyDialog.kt

```diff
@@ -18,15 +27,52 @@ import org.wikipedia.history.HistoryEntry
 import org.wikipedia.settings.Prefs
 import org.wikipedia.util.DimenUtil
 import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.UriUtil
 
 object SurveyDialog {
+    // @TODO: update the strings after product confirm them
+    fun showContributionSurveyDialog(
+        activity: AppCompatActivity,
+        @StringRes titleId: Int,
+        @StringRes messageId: Int,
+        @DrawableRes icon: Int? = null,
+        delayMillis: Long = 0,
+        isCancellable: Boolean = false
+    ) {
+        activity.lifecycleScope.launch {
+            val dialog = MaterialAlertDialogBuilder(activity, R.style.AlertDialogTheme_Icon)
+                .setTitle(titleId)
+                .setMessage(messageId)
+                .setCancelable(isCancellable)
+                .setPositiveButton("Take Survey") { _, _ ->
+                    UriUtil.visitInExternalBrowser(
+                        activity,
+                        Uri.parse(activity.getString(R.string.contribution_dashboard_survey_url))
+                    )
+                }
+                .setNegativeButton("No, Thanks") { _, _ ->
+                    // dismiss
```

Comment:
We will need a preference for this dialog, since we are going to show this once after saving the donor history.

Reply:
done.